homcc/server/cache.py

```diff
@@ -42,12 +83,26 @@ def _create_cache_folder(root_temp_folder: Path) -> Path:
     def get(self, hash_value: str) -> str:
         """Gets an entry (path) from the cache given a hash."""
         with self.cache_mutex:
+            self.cache.move_to_end(hash_value)
             return self.cache[hash_value]
 
     def put(self, hash_value: str, content: bytearray):
         """Stores a dependency in the cache."""
-        cached_file_path = self.cache_folder / hash_value
-        Path.write_bytes(cached_file_path, content)
+        if len(content) > self.max_size_bytes:
+            logger.error(
+                """File with hash '%s' can not be added to cache as it is larger than the maximum cache size.
+                (size in bytes: %i, max. cache size in bytes: %i)""",
+                hash,
```

Comment:
```suggestion
                hash_value,
```

Reply:
How is this not caught by our tooling? :sweat_smile:

Reply:
Good catch! I guess because there is the built-in function `hash`.